src/alphabet.rs

```diff
@@ -23,3 +23,112 @@ pub const FLICKR: &[u8; 58] = b"123456789abcdefghijkmnopqrstuvwxyzABCDEFGHJKLMNP
 /// The default alphabet used if none is given. Currently is the
 /// [`BITCOIN`](constant.BITCOIN.html) alphabet.
 pub const DEFAULT: &[u8; 58] = BITCOIN;
+
+/// Prepared Alpabet for [`EncodeBuilder`](crate::encode::EncodeBuilder) and [`DecodeBuilder`](crate::decode::DecodeBuilder).
+#[derive(Clone, Copy)]
+#[allow(missing_debug_implementations)]
+#[allow(missing_docs)]
+pub struct Alphabet {
+    pub encode: [u8; 58],
+    pub decode: [u8; 128],
+}
+
+impl Alphabet {
+    /// Create prepared alphabet.
+    pub fn new(base: &[u8; 58]) -> Alphabet {
+        assert!(base.iter().all(|&c| c < 128));
+
+        let mut encode = [0; 58];
+        encode.copy_from_slice(base);
+
+        let mut decode = [0xFF; 128];
+        for (i, &c) in base.iter().enumerate() {
+            decode[c as usize] = i as u8;
+        }
+
+        Alphabet { encode, decode }
+    }
+}
+
+/// Bitcoin's prepared alphabet.
+pub const ALPHABET_BITCOIN: &Alphabet = &Alphabet {
```

Comment:
I'd prefer these to be associated constants on `Alphabet` as shown in the playground I linked above to avoid the stuttering in the path `bs58::alphabet::ALPHABET_BITCOIN`.

Reply:
Yeah, way in attached playground looks really nice. Thanks!

Reply:
Oh, but probably makes sense to keep it as `&Self` instead of being `Self` like I had.